tests/Unit/GridLaunch/rect_tiles.cpp

```diff
@@ -15,8 +15,10 @@
 
 __attribute__((hc_grid_launch)) void kernel_call(grid_launch_parm lp, int *a_d, int pitch)
 {
-  int i = lp.groupId.x*TILE_I + lp.threadId.x;
-  int j = lp.groupId.y*TILE_J + lp.threadId.y;
+  //int i = lp.group_id.x*TILE_I + lp.thread_id.x;
+  //int j = lp.group_id.y*TILE_J + lp.thread_id.y;
+  int i = amp_get_group_id(0) * TILE_I + amp_get_local_id(0);
```

Comment:
Either will work here but I wanted the test to check that the grid_launch_parm fields for grid_dim and group_dim were being set correctly.  The interface we have designed allows the kernel to see the grid_launch_parm structure and read its fields.  We have to do a little work in the implementation to copy these fields from the host lp to the accelerator lp, and this test ensures we do those copies correctly.  Overall I think the interface is fine.  Now that LP is ‘const’ the perf impact for passing LP to accelerator will be quite small.

From: Wen-Heng (Jack) Chung [mailto:notifications@github.com]
Sent: Wednesday, June 22, 2016 8:11 PM
To: RadeonOpenCompute/hcc <hcc@noreply.github.com>
Cc: Sander, Ben <ben.sander@amd.com>; Author <author@noreply.github.com>
Subject: Re: [RadeonOpenCompute/hcc] Grid launch 20 a (#80)


In tests/Unit/GridLaunch/rect_tiles.cpp<https://github.com/RadeonOpenCompute/hcc/pull/80#discussion_r68162394>:

> @@ -15,8 +15,10 @@

>

>  __attribute__((hc_grid_launch)) void kernel_call(grid_launch_parm lp, int *a_d, int pitch)

>  {

> -  int i = lp.groupId.x*TILE_I + lp.threadId.x;

> -  int j = lp.groupId.y*TILE_J + lp.threadId.y;

> +  //int i = lp.group_id.x*TILE_I + lp.thread_id.x;

> +  //int j = lp.group_id.y*TILE_J + lp.thread_id.y;

> +  int i = amp_get_group_id(0) * TILE_I + amp_get_local_id(0);

How about use "hc_get_group_id()" and "hc_get_workitem_id()"?

—
You are receiving this because you authored the thread.
Reply to this email directly, view it on GitHub<https://github.com/RadeonOpenCompute/hcc/pull/80/files/758b7f57d57a3a927dd454b7d8011318ecb129f0#r68162394>, or mute the thread<https://github.com/notifications/unsubscribe/ACYSArP8jJzhA-ezW1bluwlFZy1I1RDLks5qOd0tgaJpZM4I8LAJ>.